<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MargheRita: an R package for LC-MS/MS SWATH metabolomics data analysis and confident metabolite identification based on a spectral library of reference standards • margheRita</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MargheRita: an R package for LC-MS/MS SWATH metabolomics data analysis and confident metabolite identification based on a spectral library of reference standards">
<meta property="og:description" content="margheRita">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">margheRita</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/margheRita.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MargheRita: an R package for LC-MS/MS SWATH
metabolomics data analysis and confident metabolite identification based
on a spectral library of reference standards</h1>
            
      
      
      <div class="hidden name"><code>margheRita.Rmd</code></div>

    </div>

    
    
<p><img src="images/logo.png" style="width:30.0%"></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Untargeted metabolomics allows acquisition of thousands metabolite
signals in a single sample that demands computational techniques for
post-acquisition steps. Web-based data processing solutions as well as R
packages exist and provide tools for metabolomics data analysis.
Frequently these tools cover only a part of the entire workflow thus
requiring the use of different platforms. Moreover, despite the
existence of several databases, metabolite identification remains the
bottleneck in metabolomics due to the high variability in the
fragmentation pattern resulting from different mass spectrometer
configuration-based libraries.</p>
<p>margheRita covers the whole data analysis workflow in LC-MS/MS
untargeted metabolomics experiments, using MS and MS/MS (even SWATH)
data. It takes in input the results of data extraction generated by <a href="http://prime.psc.riken.jp/compms/msdial/main.html" class="external-link">MS-Dial</a>
<span class="citation">(<a href="#ref-Tsugawa2015" role="doc-biblioref">Tsugawa et al. 2015</a>)</span> and metadata for
sample processing (in text or Excel format).</p>
<p>The package provides:</p>
<ul>
<li>a series of pre-processing functions (quality control, filtering and
normalization) with a particular focus on methods specifically
recommended for metabolomic profiles, such as filtering by mass defects,
filtering by coefficient of variation (samples vs QCs) and probabilistic
quotient normalization;</li>
<li>metabolite annotation up to level-1, based on in-house spectral
libraries as well as freely available libraries;</li>
<li>spectral libraries that covers 4 different chromatographic column
types: RP-C18, HILIC, RP-C8 and pZIC-HILIC Zwitterionic.</li>
<li>simplified execution of parametric and non-parametric statistical
tests over a large number of features;</li>
<li>pathway analysis based on ORA and MSEA over various databases.</li>
</ul>
<p>Contacts:</p>
<ul>
<li>
<a href="https://research.hsr.it/en/core-facilities/promefa/annapaola-andolfo.html" class="external-link">Annapaola
Andolfo</a>, Proteomics and Metabolomcis Facility, HSR</li>
<li>
<a href="https://www.itb.cnr.it/en/institute/staff/ettore-mosca" class="external-link">Ettore
Mosca</a>, Bioinformatics Lab, CNR-ITB</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package requires a series of other R packages, which are availble
in <a href="https://cran.r-project.org/" class="external-link">CRAN</a>, <a href="https://www.bioconductor.org/" class="external-link">Bioconductor</a> or<a href="https://github.com/" class="external-link">github</a>. In most of the cases, the
following instructions guarantee that all such dependencies are
installed:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pcastellanoescuder/POMA"</span>, <span class="st">"antonvsdata/notame"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"clusterProfiler"</span>, <span class="st">"pcaMethods"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"emosca-cnr/margheRita"</span>, dependencies <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<p>margheRita is intended to be used after having done a number of data
acquisition steps through MS-Dial <span class="citation">(<a href="#ref-Tsugawa2015" role="doc-biblioref">Tsugawa et al.
2015</a>)</span>. It requires two text files in tab-delimited
format:</p>
<ul>
<li>
<strong>feature data file</strong>, which must include feature
identifiers, m/z values, retention times, MS/MS spectra and feature
abundances across samples; usually, this file is generated by
MS-Dial;</li>
<li>
<strong>sample annotation file</strong>, which must include the
following mandatory columns: “id”, “injection_order”, “batch”, “class”,
“biological_rep” and “technical_rep”.</li>
</ul>
<p>Usingh such two files, the function <code><a href="../reference/read_input_file.html">read_input_file()</a></code>
creates the “mRList” object, which is used by most of the margheRita
functions as main input/output:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_input_file.html">read_input_file</a></span><span class="op">(</span>feature_file <span class="op">=</span> <span class="st">"MS-Dial_file.txt"</span>, sample_file <span class="op">=</span> <span class="st">"sample_info.txt"</span><span class="op">)</span></span></code></pre></div>
<p>Please, also consider that, to properly split QC samples, you must
indicate exactly “QC” as class of a sample in the sample_file.</p>
<p>The initial mRList object contains the following elements:</p>
<table class="table">
<thead><tr class="header">
<th>Element of the mRList</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>data</td>
<td>matrix containing metabolite abundances</td>
</tr>
<tr class="even">
<td>metab_ann</td>
<td>metabolite annotation</td>
</tr>
<tr class="odd">
<td>sample_ann</td>
<td>sample annotation</td>
</tr>
<tr class="even">
<td>QC</td>
<td>matrix containing metabolite abundances of QC samples</td>
</tr>
<tr class="odd">
<td>QC_ann</td>
<td>annotation of QC samples</td>
</tr>
</tbody>
</table>
<pre><code><span><span class="co">##      x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## F248                                            294.9630</span></span>
<span><span class="co">## F260                                            299.8641</span></span>
<span><span class="co">## F338                                            355.7661</span></span>
<span><span class="co">## F506                                            108.1681</span></span>
<span><span class="co">## F507                                            462.1838</span></span>
<span><span class="co">## F511                                          52674.7600</span></span>
<span><span class="co">##      x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## F248                                            307.2657</span></span>
<span><span class="co">## F260                                            350.1306</span></span>
<span><span class="co">## F338                                            332.5398</span></span>
<span><span class="co">## F506                                            141.9220</span></span>
<span><span class="co">## F507                                            466.3308</span></span>
<span><span class="co">## F511                                          79656.2100</span></span>
<span><span class="co">##      x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## F248                                            157.4713</span></span>
<span><span class="co">## F260                                            663.3356</span></span>
<span><span class="co">## F338                                            457.2227</span></span>
<span><span class="co">## F506                                            314.8479</span></span>
<span><span class="co">## F507                                            810.1950</span></span>
<span><span class="co">## F511                                          24772.5900</span></span></code></pre>
<pre><code><span><span class="co">##      Feature_ID                                                MSDialName</span></span>
<span><span class="co">## F248       F248                                                   Unknown</span></span>
<span><span class="co">## F260       F260                                                   Unknown</span></span>
<span><span class="co">## F338       F338                                                   Unknown</span></span>
<span><span class="co">## F506       F506                                                   Unknown</span></span>
<span><span class="co">## F507       F507                                                   Unknown</span></span>
<span><span class="co">## F511       F511 Trimethylamine N-oxide; CE30; UYPYRKYUKCHHIB-UHFFFAOYSA-N</span></span>
<span><span class="co">##      MSDialSMILES    rt       mz</span></span>
<span><span class="co">## F248         null 0.890 61.63427</span></span>
<span><span class="co">## F260         null 0.944 61.70882</span></span>
<span><span class="co">## F338         null 3.677 67.05233</span></span>
<span><span class="co">## F506         null 1.260 76.03783</span></span>
<span><span class="co">## F507         null 0.868 76.03820</span></span>
<span><span class="co">## F511   CN(=O)(C)C 0.888 76.07437</span></span></code></pre>
<pre><code><span><span class="co">##                                                                                                      id</span></span>
<span><span class="co">## x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">## x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">##                                                     injection_order batch class</span></span>
<span><span class="co">## x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos              14     1    DD</span></span>
<span><span class="co">## x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos              15     1    MM</span></span>
<span><span class="co">## x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos              16     1    AA</span></span>
<span><span class="co">## x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos              17     1    AA</span></span>
<span><span class="co">## x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos              18     1    DD</span></span>
<span><span class="co">## x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos              19     1    MM</span></span>
<span><span class="co">##                                                     technical_rep</span></span>
<span><span class="co">## x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos             1</span></span>
<span><span class="co">## x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos             1</span></span>
<span><span class="co">## x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos             1</span></span>
<span><span class="co">## x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos             1</span></span>
<span><span class="co">## x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos             1</span></span>
<span><span class="co">## x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos             1</span></span>
<span><span class="co">##                                                     biological_rep</span></span>
<span><span class="co">## x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos      mealC_t07</span></span>
<span><span class="co">## x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos      mealB_t00</span></span>
<span><span class="co">## x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos      mealA_t00</span></span>
<span><span class="co">## x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos      mealC_t01</span></span>
<span><span class="co">## x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos      mealA_t06</span></span>
<span><span class="co">## x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos      mealC_t08</span></span>
<span><span class="co">##                                                     subj_meal_time  meal time</span></span>
<span><span class="co">## x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos   DD_mealC_t07 mealC  t07</span></span>
<span><span class="co">## x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos   MM_mealB_t00 mealB  t00</span></span>
<span><span class="co">## x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos   AA_mealA_t00 mealA  t00</span></span>
<span><span class="co">## x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos   AA_mealC_t01 mealC  t01</span></span>
<span><span class="co">## x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos   DD_mealA_t06 mealA  t06</span></span>
<span><span class="co">## x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos   MM_mealC_t08 mealC  t08</span></span>
<span><span class="co">##                                                     subj_meal</span></span>
<span><span class="co">## x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos  DD_mealC</span></span>
<span><span class="co">## x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos  MM_mealB</span></span>
<span><span class="co">## x016_UR_norm_AA_mealA_t00_rtech1_RP_short_SWATH_pos  AA_mealA</span></span>
<span><span class="co">## x017_UR_norm_AA_mealC_t01_rtech1_RP_short_SWATH_pos  AA_mealC</span></span>
<span><span class="co">## x018_UR_norm_DD_mealA_t06_rtech1_RP_short_SWATH_pos  DD_mealA</span></span>
<span><span class="co">## x019_UR_norm_MM_mealC_t08_rtech1_RP_short_SWATH_pos  MM_mealC</span></span></code></pre>
<pre><code><span><span class="co">##      x024_QC_norm_06_RP_short_SWATH_pos x048_QC_norm_10_RP_short_SWATH_pos</span></span>
<span><span class="co">## F248                           89.55031                          137.85050</span></span>
<span><span class="co">## F260                          225.45610                          134.20190</span></span>
<span><span class="co">## F338                          303.82490                          180.66890</span></span>
<span><span class="co">## F506                          163.48670                           91.28957</span></span>
<span><span class="co">## F507                          412.40590                          187.70180</span></span>
<span><span class="co">## F511                        39594.32000                        22377.21000</span></span>
<span><span class="co">##      x072_QC_norm_14_RP_short_SWATH_pos</span></span>
<span><span class="co">## F248                           142.8374</span></span>
<span><span class="co">## F260                           153.3819</span></span>
<span><span class="co">## F338                           172.7377</span></span>
<span><span class="co">## F506                           140.5623</span></span>
<span><span class="co">## F507                           408.4307</span></span>
<span><span class="co">## F511                         26902.7900</span></span></code></pre>
<pre><code><span><span class="co">##                                                                    id</span></span>
<span><span class="co">## x024_QC_norm_06_RP_short_SWATH_pos x024_QC_norm_06_RP_short_SWATH_pos</span></span>
<span><span class="co">## x048_QC_norm_10_RP_short_SWATH_pos x048_QC_norm_10_RP_short_SWATH_pos</span></span>
<span><span class="co">## x072_QC_norm_14_RP_short_SWATH_pos x072_QC_norm_14_RP_short_SWATH_pos</span></span>
<span><span class="co">## x119_QC_norm_21_RP_short_SWATH_pos x119_QC_norm_21_RP_short_SWATH_pos</span></span>
<span><span class="co">## x191_QC_norm_33_RP_short_SWATH_pos x191_QC_norm_33_RP_short_SWATH_pos</span></span>
<span><span class="co">## x204_QC_norm_36_RP_short_SWATH_pos x204_QC_norm_36_RP_short_SWATH_pos</span></span>
<span><span class="co">##                                    injection_order batch class technical_rep</span></span>
<span><span class="co">## x024_QC_norm_06_RP_short_SWATH_pos              24     1    QC             6</span></span>
<span><span class="co">## x048_QC_norm_10_RP_short_SWATH_pos              48     1    QC            10</span></span>
<span><span class="co">## x072_QC_norm_14_RP_short_SWATH_pos              72     1    QC            14</span></span>
<span><span class="co">## x119_QC_norm_21_RP_short_SWATH_pos             119     1    QC            21</span></span>
<span><span class="co">## x191_QC_norm_33_RP_short_SWATH_pos             191     1    QC            33</span></span>
<span><span class="co">## x204_QC_norm_36_RP_short_SWATH_pos             204     2    QC            36</span></span>
<span><span class="co">##                                    biological_rep subj_meal_time meal time</span></span>
<span><span class="co">## x024_QC_norm_06_RP_short_SWATH_pos             QC          QC_QC   QC   QC</span></span>
<span><span class="co">## x048_QC_norm_10_RP_short_SWATH_pos             QC          QC_QC   QC   QC</span></span>
<span><span class="co">## x072_QC_norm_14_RP_short_SWATH_pos             QC          QC_QC   QC   QC</span></span>
<span><span class="co">## x119_QC_norm_21_RP_short_SWATH_pos             QC          QC_QC   QC   QC</span></span>
<span><span class="co">## x191_QC_norm_33_RP_short_SWATH_pos             QC          QC_QC   QC   QC</span></span>
<span><span class="co">## x204_QC_norm_36_RP_short_SWATH_pos             QC          QC_QC   QC   QC</span></span>
<span><span class="co">##                                    subj_meal</span></span>
<span><span class="co">## x024_QC_norm_06_RP_short_SWATH_pos     QC_QC</span></span>
<span><span class="co">## x048_QC_norm_10_RP_short_SWATH_pos     QC_QC</span></span>
<span><span class="co">## x072_QC_norm_14_RP_short_SWATH_pos     QC_QC</span></span>
<span><span class="co">## x119_QC_norm_21_RP_short_SWATH_pos     QC_QC</span></span>
<span><span class="co">## x191_QC_norm_33_RP_short_SWATH_pos     QC_QC</span></span>
<span><span class="co">## x204_QC_norm_36_RP_short_SWATH_pos     QC_QC</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="inter-operability">Inter-operability<a class="anchor" aria-label="anchor" href="#inter-operability"></a>
</h2>
<p>To support interoperability with other packages with a focus on
metabolomics, the margheRitaList can be reorganized as a “MetaboSet”
object, used by package “notame” <span class="citation">(<a href="#ref-Klavus2020" role="doc-biblioref">Klåvus et al.
2020</a>)</span>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/as.metaboset.html">as.metaboset</a></span><span class="op">(</span><span class="va">mRList</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MetaboSet object with 290 features and 253 samples.</span></span>
<span><span class="co">## 10 QC samples included</span></span>
<span><span class="co">## 290 non-flagged features, 0 flagged features.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## class:</span></span>
<span><span class="co">##    DD: 81, MM: 81, AA: 81, QC: 10 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The object has the following parts (splits):</span></span>
<span><span class="co">##   FALSE:  features</span></span></code></pre>
<p>or as “PomaSummarizedExperiment” object, used by package “POMA” <span class="citation">(<a href="#ref-Castellano2021" role="doc-biblioref">Castellano-Escuder et al. 2021</a>)</span>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/as.PomaSummarizedExperiment.html">as.PomaSummarizedExperiment</a></span><span class="op">(</span><span class="va">mRList</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 290 253 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(290): F338 F506 ... F41874 F42145</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(253): x014_UR_norm_DD_mealC_t07_rtech1_RP_short_SWATH_pos</span></span>
<span><span class="co">##   x015_UR_norm_MM_mealB_t00_rtech1_RP_short_SWATH_pos ...</span></span>
<span><span class="co">##   x286_QC_norm_49_RP_short_SWATH_pos x295_QC_norm_51_RP_short_SWATH_pos</span></span>
<span><span class="co">## colData names(10): class biological_rep ... injection_order</span></span>
<span><span class="co">##   technical_rep</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="filtering-imputation-and-normalization">Filtering, imputation and normalization<a class="anchor" aria-label="anchor" href="#filtering-imputation-and-normalization"></a>
</h2>
<p>The function <code><a href="../reference/filtering.html">filtering()</a></code> runs filters to exclude
features/sample with many missing values, features with wrong m/z values
and, lastly, performs imputation of missing values:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filtering.html">filtering</a></span><span class="op">(</span><span class="va">mL</span><span class="op">)</span></span></code></pre></div>
<p>These three steps can be called independently through the function
<code><a href="../reference/filter_NA.html">filter_NA()</a></code>, <code><a href="../reference/m_z_filtering.html">m_z_filtering()</a></code> and
<code><a href="../reference/imputation.html">imputation()</a></code>, respectively. In particular,
<code><a href="../reference/m_z_filtering.html">m_z_filtering()</a></code> remove features with m/z that have decimal
value within [4, 8] (by default), while the imputation is performed
replacing NA values with a random number, calculated between 10%-25% of
the minimum value of the feature.</p>
<p>The function <code><a href="../reference/heatscatter_chromatography.html">heatscatter_chromatography()</a></code> creates a
graphic overview of the mz and rt values in the dataset:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/heatscatter_chromatography.html">heatscatter_chromatography</a></span><span class="op">(</span><span class="va">mL</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/hs.png" style="width:100.0%"></p>
<p>margheRita provides three ways for normalizing metabolite
profiles:</p>
<ul>
<li>“log”, the log2 of metabolite abundances;</li>
<li>“reference”, every sample is divided by a reference value;</li>
<li>“pqn”, probabilistic quotient normalization <span class="citation">(<a href="#ref-Dieterle2006" role="doc-biblioref">Dieterle et al. 2006</a>)</span>;</li>
</ul>
<p>For “reference” and “pqn” methods, the column <code>reference</code>
must be present in <code>mRList$metab_ann</code>. If missing, the
function <code><a href="../reference/calc_reference.html">calc_reference()</a></code> sets up such column using average
metabolite values and medians of QC samples. For example, here’s a call
to <code><a href="../reference/normalize_profiles.html">normalize_profiles()</a></code> using pqn:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_profiles.html">normalize_profiles</a></span><span class="op">(</span><span class="va">mL</span>, method <span class="op">=</span> <span class="st">"pqn"</span><span class="op">)</span></span></code></pre></div>
<p>The comparison of the coefficient of variation of a metabolite in
relation to QC samples provides a means to exclude low quality features.
In particular, only features that have a CV ratio between no-QC samples
and QC sample higher than a given threshold (by default 1) are kept:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CV_ratio.html">CV_ratio</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm</span><span class="op">)</span></span></code></pre></div>
<p>The distributions of metabolite relative log-abundances can be
calculated and visualized by means of:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RLA.html">RLA</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL</span><span class="op">)</span></span></code></pre></div>
<p>Typically, after normalization, the various samples should have
similar distributions of relative log-abundances. <img src="images/rla.png" style="width:100.0%"></p>
</div>
<div class="section level2">
<h2 id="principal-component-analysis">Principal Component Analysis<a class="anchor" aria-label="anchor" href="#principal-component-analysis"></a>
</h2>
<p>margheRita performs Principal Component Analysis (PCA) using the
function <code><a href="../reference/mR_pca.html">mR_pca()</a></code>, which relies on the package pca_methods
<span class="citation">(<a href="#ref-Stacklies2007" role="doc-biblioref">Stacklies et al. 2007</a>)</span>. Besides choosing
the scaling method (argument <code>scaling</code>) and number of PCs
(<code>nPcS</code>), it allows to include/exclude quality control
samples by means of argument <code>include_QC</code>:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mR_pca.html">mR_pca</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm</span>, nPcs<span class="op">=</span><span class="fl">5</span>, scaling<span class="op">=</span><span class="st">"uv"</span>, include_QC<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The results are added to the mRList in the element <code>pca</code>.
It also provides some plots, like the visualization of distribution of
loadings for all-pairs of the top 5 PCs. The plots are directly saved in
the current working directory (or in the sub directory created with the
argument <code>dirout</code>)</p>
<p><img src="images/pairs.png" style="width:100.0%"> The results of
PCA can be plotted using <code><a href="../reference/Plot2DPCA.html">Plot2DPCA()</a></code> function. The argument
<code>col_by</code> enables the choice of the
<code>mRList$sample_ann</code> column to be used to color samples:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Plot2DPCA.html">Plot2DPCA</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm</span>, pcx<span class="op">=</span><span class="fl">1</span>, pcy<span class="op">=</span><span class="fl">2</span>, col_by<span class="op">=</span><span class="st">"class"</span>, include_QC<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/scores.png" style="width:80.0%"></p>
</div>
<div class="section level2">
<h2 id="removing-samples-and-collapsing-techincal-replicates">Removing samples and collapsing techincal replicates<a class="anchor" aria-label="anchor" href="#removing-samples-and-collapsing-techincal-replicates"></a>
</h2>
<p>It is common that the inspection of the similarity between samples
(e.g. distribution over the top PCs, RLA) rise concerns about the
quality of some samples. The function <code><a href="../reference/remove_samples.html">remove_samples()</a></code>
allows the user to remove one or more samples from the
<code>mRList</code>. Here, for example we remove all “Blank”
samples:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_samples.html">remove_samples</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL</span>, ids <span class="op">=</span> <span class="st">"Blank"</span>, column <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span></code></pre></div>
<p>In this case, the function removes all samples with value “Blank” in
the column “class” of sample annotation.</p>
<p>The definition of mean metabolite abundance for every biological
replicate is performed by means of <code><a href="../reference/collapse_tech_rep.html">collapse_tech_rep()</a></code>
function:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL_norm_bio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/collapse_tech_rep.html">collapse_tech_rep</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       AA_mealA_t00 AA_mealA_t01 AA_mealA_t02 AA_mealA_t03 AA_mealA_t04</span></span>
<span><span class="co">## F338      293.7091     562.2148     523.8581     356.1212     757.9913</span></span>
<span><span class="co">## F506      359.8439     300.3907     637.1368     323.6169     191.9161</span></span>
<span><span class="co">## F513      660.1863     557.7189     571.6556     510.2219     453.0921</span></span>
<span><span class="co">## F966    11301.1699    8998.2739    9584.4802    7777.3442    7946.9138</span></span>
<span><span class="co">## F1016    3587.4424    2957.3787    4063.9745    2520.4270    2633.2531</span></span>
<span><span class="co">## F1039    1162.8266    1273.1433    1812.5802    2206.6794    3086.8158</span></span></code></pre>
<pre><code><span><span class="co">##              class_biorep class biological_rep</span></span>
<span><span class="co">## AA_mealA_t00 AA_mealA_t00    AA      mealA_t00</span></span>
<span><span class="co">## AA_mealA_t01 AA_mealA_t01    AA      mealA_t01</span></span>
<span><span class="co">## AA_mealA_t02 AA_mealA_t02    AA      mealA_t02</span></span>
<span><span class="co">## AA_mealA_t03 AA_mealA_t03    AA      mealA_t03</span></span>
<span><span class="co">## AA_mealA_t04 AA_mealA_t04    AA      mealA_t04</span></span>
<span><span class="co">## AA_mealA_t05 AA_mealA_t05    AA      mealA_t05</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="statistical-analysis">Statistical analysis<a class="anchor" aria-label="anchor" href="#statistical-analysis"></a>
</h2>
<p>MargheRita provides some functions to calculate mean and variability,
fold changes and to test for metabolite variations.</p>
<p>The function <code><a href="../reference/mean_median_stdev_samples.html">mean_median_stdev_samples()</a></code> calculates
mean, median and standard deviation of metabolite abundance according to
the sample classes specified in the column “class” of sample
annotation:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/mean_median_stdev_samples.html">mean_median_stdev_samples</a></span><span class="op">(</span><span class="va">mL_norm_bio</span><span class="op">)</span></span></code></pre></div>
<p>The function <code><a href="../reference/univariate.html">univariate()</a></code> performs dataset-wide
statistical tests (Student t-tests, Wilcoxon test, Anova and
Kruskal-Wallis test) between levels of a particular factor defined in
the sample annotation:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL_norm_bio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/univariate.html">univariate</a></span><span class="op">(</span><span class="va">mL_norm_bio</span>, test_method<span class="op">=</span><span class="st">"anova"</span>, exp.levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DD"</span>, <span class="st">"MM"</span><span class="op">)</span>, exp.factor <span class="op">=</span> <span class="st">"class"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                F            p            q       DD-AA        MM-AA</span></span>
<span><span class="co">## F338  17.6282737 4.821921e-07 2.497066e-06 0.003610291 2.425930e-07</span></span>
<span><span class="co">## F506  57.2151868 5.070120e-16 8.168526e-15 0.000000000 0.000000e+00</span></span>
<span><span class="co">## F513  11.7011315 3.596096e-05 1.271790e-04 0.073328306 1.960049e-05</span></span>
<span><span class="co">## F966   9.8629476 1.518144e-04 4.359026e-04 0.939935238 1.281721e-03</span></span>
<span><span class="co">## F1016 16.8591321 8.219521e-07 4.181861e-06 0.035099636 4.686136e-03</span></span>
<span><span class="co">## F1039  0.1269624 8.809485e-01 9.091639e-01 0.897133234 9.998608e-01</span></span>
<span><span class="co">##              MM-DD</span></span>
<span><span class="co">## F338  3.137240e-02</span></span>
<span><span class="co">## F506  7.838339e-01</span></span>
<span><span class="co">## F513  2.905768e-02</span></span>
<span><span class="co">## F966  4.121310e-04</span></span>
<span><span class="co">## F1016 4.167037e-07</span></span>
<span><span class="co">## F1039 9.039911e-01</span></span></code></pre>
<p>The function <code><a href="../reference/metab_boxplot.html">metab_boxplot()</a></code> draws boxplots of feature
abundances grouped by the levels of a given factor:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/metab_boxplot.html">metab_boxplot</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm_bio</span>, col_by<span class="op">=</span><span class="st">"class"</span>, group<span class="op">=</span><span class="st">"class"</span>, features <span class="op">=</span> <span class="st">"F3957"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/boxplot.F3957.png" style="width:80.0%"></p>
<p>This function allows to draw boxplot and report p-value calculated
using Student t test. The function <code><a href="../reference/h_map.html">h_map()</a></code> provides
heatmaps based on package ComplexHeatmap <span class="citation">(<a href="#ref-Gu2016" role="doc-biblioref">Gu, Eils, and Schlesner
2016</a>)</span>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/h_map.html">h_map</a></span><span class="op">(</span><span class="va">mL_norm_bio</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/hm.png" style="width:80.0%"></p>
</div>
<div class="section level2">
<h2 id="metabolite-identification">Metabolite identification<a class="anchor" aria-label="anchor" href="#metabolite-identification"></a>
</h2>
<p>Metabolite identification in margheRita is performed by means of the
function <code><a href="../reference/metabolite_identification.html">metabolite_identification()</a></code>, which requires an
mRList object and a reference library with MS and MS/MS metabolite
information. The identification is possible up to level-1, provided that
the required information are available in the reference library. The
identification is based on the quantification of the following
quantities:</p>
<ul>
<li>retention time (RT) error: <span class="math display">\[\epsilon_t(i) = |t(i) - t^*(i)|\]</span>
</li>
<li>ppm error: <span class="math display">\[\epsilon_m(i) = \frac{|m(i)
- m^*(i)|}{m^*(i)} \cdot 10^6\]</span>
</li>
<li>percent relative intensity error: <span class="math display">\[\epsilon_{I_R}(i,j) = \frac{|I_R(i,j) -
I^*(i,j)|}{I^*_R(i,j)} \cdot 100\]</span>
</li>
</ul>
<p>Such quantities are used to score the similarity among precursors of
features and metabolites, as well as their MS/MS spectra.</p>
<p>The function <code><a href="../reference/select_library.html">select_library()</a></code> provides a means to select
any of two sources:</p>
<ul>
<li><p><strong>margheRita</strong>, which contains MS and MS/MS
information for about 800 metabolites spanning several biological
functions; these libraries provideup to level 1 identifications in
positive and negative modalities for “HILIC”, “LipC8”, “pZIC”, “RPLong”
and “RPShort” chromatographic columns, that are acquired following the
methods reported in the supplementary material of Mosca et
al. (manuscript in preparation).</p></li>
<li><p><strong>MS-Dial</strong>, which covers a much larger set of
metabolites (<span class="math inline">\(10^5\)</span>), but is limited
to level 2 identifications in positive and negative modalities.</p></li>
</ul>
<p>In this example, we load the margheRita library in positive
modalitity with retention times of RPShort columns and we discard all
peaks with relative intensity less than 10:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mR_library</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_library.html">select_library</a></span><span class="op">(</span>column <span class="op">=</span> <span class="st">"RPShort"</span>, mode <span class="op">=</span> <span class="st">"POS"</span>, accept_RI<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>The resulting <code>mR_library</code> is a list that contains
information about precursors</p>
<pre><code><span><span class="co">##      ID      CAS                         Name   rt        mz PubChemCID</span></span>
<span><span class="co">## L10 L10 485-80-3 S-(5'-Adenosyl)-L-methionine 0.80 399.14452      34756</span></span>
<span><span class="co">## L14 L14  61-19-8      Adenosine monophosphate 1.40 348.07037       6083</span></span>
<span><span class="co">## L17 L17 979-92-0       S-Adenosylhomocysteine 1.28 385.12887     439155</span></span>
<span><span class="co">## L20 L20  56-86-0              L-Glutamic acid 0.90 148.06044      33032</span></span>
<span><span class="co">## L30 L30  56-40-6                      Glycine 0.85  76.03931        750</span></span>
<span><span class="co">## L33 L33  56-41-7                    L-Alanine 0.88  90.05496       5950</span></span>
<span><span class="co">##                                                                       SMILES</span></span>
<span><span class="co">## L10 C[S+](CC[C@H](N)C(O)=O)C[C@H]1O[C@H]([C@H](O)[C@@H]1O)N1C=NC2=C1N=CN=C2N</span></span>
<span><span class="co">## L14             NC1=NC=NC2=C1N=CN2[C@@H]1O[C@H](COP(O)(O)=O)[C@@H](O)[C@H]1O</span></span>
<span><span class="co">## L17    N[C@@H](CCSC[C@H]1O[C@H]([C@H](O)[C@@H]1O)N1C=NC2=C(N)N=CN=C12)C(O)=O</span></span>
<span><span class="co">## L20                                                  N[C@@H](CCC(O)=O)C(O)=O</span></span>
<span><span class="co">## L30                                                                 NCC(O)=O</span></span>
<span><span class="co">## L33                                                          C[C@H](N)C(O)=O</span></span></code></pre>
<p>and MS/MS peaks</p>
<pre><code><span><span class="co">## $M1</span></span>
<span><span class="co">##           [,1]      [,2]</span></span>
<span><span class="co">## [1,]  45.03236  10.34602</span></span>
<span><span class="co">## [2,]  70.02762  12.46120</span></span>
<span><span class="co">## [3,]  96.00721  13.50730</span></span>
<span><span class="co">## [4,]  99.04249  17.68019</span></span>
<span><span class="co">## [5,] 113.03337 100.00000</span></span>
<span><span class="co">## [6,] 117.05313  20.43913</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $M2</span></span>
<span><span class="co">##            [,1]      [,2]</span></span>
<span><span class="co">##  [1,]  57.03256  11.18379</span></span>
<span><span class="co">##  [2,]  60.07970  58.46973</span></span>
<span><span class="co">##  [3,]  85.02673 100.00000</span></span>
<span><span class="co">##  [4,]  95.08432  12.06547</span></span>
<span><span class="co">##  [5,] 109.09995  11.77031</span></span>
<span><span class="co">##  [6,] 144.10080  38.14543</span></span>
<span><span class="co">##  [7,] 183.17293  27.40629</span></span>
<span><span class="co">##  [8,] 285.20408  26.57983</span></span>
<span><span class="co">##  [9,] 344.27686  33.27335</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $M3</span></span>
<span><span class="co">##           [,1]      [,2]</span></span>
<span><span class="co">## [1,]  55.01692 100.00000</span></span>
<span><span class="co">## [2,]  56.04866  58.49046</span></span>
<span><span class="co">## [3,]  70.06436  54.29821</span></span>
<span><span class="co">## [4,]  98.05859  90.19182</span></span>
<span><span class="co">## [5,] 116.06901  16.49188</span></span></code></pre>
<p>Once the library is selected, metabolite identification can be
performed by the homonymous function, where the argument
<code>features</code> specifies the features to be considered (all
features if it is left <code>features=NULL</code>, as in the following
example):</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mL_norm_bio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/metabolite_identification.html">metabolite_identification</a></span><span class="op">(</span><span class="va">mL_norm_bio</span>, library_list <span class="op">=</span> <span class="va">mR_library</span><span class="op">)</span></span></code></pre></div>
<p>The function <code><a href="../reference/metabolite_identification.html">metabolite_identification()</a></code> has a series of
parameters that can be adjusted to optimize the identification process
(see its documentation). By default, all association that met the
considered criteria are returned. When metabolite identification is
applied on a large number of features (e.g., <span class="math inline">\(10^3\)</span>), it’s common to obtain multiple
features associated with the same metabolite and the opposite (1
feature, multiple metabolites). This redundancy can be addressed setting
<code>filter=TRUE</code>. In this case, the various features associated
with the same metabolite are filtered considering the classification
(Level 1, Level 2, Level 3a and Level 3b), the errors (see above) and a
series of quantitative and qualitative scores (see below and our article
for further details). The resulting associations (including full details
of the analysis) are stored in
<code>mL_norm_bio$metabolite_identification$associations</code>:</p>
<pre><code><span><span class="co">##     Feature_ID    rt       mz RT_err     RT_class RT_flag ppm_error mass_flag</span></span>
<span><span class="co">## 553     F14798 2.931 282.1190  0.069        super    TRUE 2.3543199      TRUE</span></span>
<span><span class="co">## 544      F6378 1.217 182.0811  0.413        super    TRUE 0.4029412      TRUE</span></span>
<span><span class="co">## 137     F16399 5.698 299.1265  1.272 unacceptable   FALSE 4.4438533      TRUE</span></span>
<span><span class="co">## 325      F4004 1.316 148.0601  0.416        super    TRUE 2.1472853      TRUE</span></span>
<span><span class="co">## 590      F9130 1.708 218.1370  0.658   acceptable    TRUE 7.6480152      TRUE</span></span>
<span><span class="co">##     mass_status ID_peaks peaks_found_ppm_RI matched_peaks_ratio</span></span>
<span><span class="co">## 553       super    M1758                  1           1.0000000</span></span>
<span><span class="co">## 544       super    M1569                  6           0.8571429</span></span>
<span><span class="co">## 137       super    M1110                  5           0.7142857</span></span>
<span><span class="co">## 325       super    M1592                  2           0.6666667</span></span>
<span><span class="co">## 590  acceptable     M505                  2           0.6666667</span></span>
<span><span class="co">##     precursor_in_MSMS    ID                  Name rt_lib   mz_lib</span></span>
<span><span class="co">## 553             FALSE  L801     1-Methyladenosine   3.00 282.1197</span></span>
<span><span class="co">## 544             FALSE   L77            L-Tyrosine   1.63 182.0812</span></span>
<span><span class="co">## 137             FALSE L1352         Enterolactone   6.97 299.1278</span></span>
<span><span class="co">## 325             FALSE   L20       L-Glutamic acid   0.90 148.0604</span></span>
<span><span class="co">## 590             FALSE  L906 Propionyl-L-carnitine   1.05 218.1387</span></span>
<span><span class="co">##                                                     SMILES Level Level_note</span></span>
<span><span class="co">## 553 CN1C=NC2=C(N=CN2[C@@H]2O[C@H](CO)[C@@H](O)[C@H]2O)C1=N     1           </span></span>
<span><span class="co">## 544                         N[C@@H](CC1=CC=C(O)C=C1)C(O)=O     1           </span></span>
<span><span class="co">## 137          C1C(C(C(=O)O1)CC2=CC(=CC=C2)O)CC3=CC(=CC=C3)O     2           </span></span>
<span><span class="co">## 325                                N[C@@H](CCC(O)=O)C(O)=O     1           </span></span>
<span><span class="co">## 590                      CCC(=O)OC(CC([O-])=O)C[N+](C)(C)C     1</span></span></code></pre>
<p>A summary of the associations is available in
<code>mL_norm_bio$metabolite_identification$associations_summary</code>:</p>
<pre><code><span><span class="co">##     Feature_ID    ID                  Name Level Level_note</span></span>
<span><span class="co">## 553     F14798  L801     1-Methyladenosine     1           </span></span>
<span><span class="co">## 544      F6378   L77            L-Tyrosine     1           </span></span>
<span><span class="co">## 137     F16399 L1352         Enterolactone     2           </span></span>
<span><span class="co">## 325      F4004   L20       L-Glutamic acid     1           </span></span>
<span><span class="co">## 590      F9130  L906 Propionyl-L-carnitine     1</span></span></code></pre>
<p>Lastly, the associations are used to add metabolite information to
<code>mL_norm_bio$metab_ann</code> (here, we omit MS/MS spectra for the
sake of brevity):</p>
<pre><code><span><span class="co">##   Feature_ID                                                  MSDialName</span></span>
<span><span class="co">## 1       F506                                                     Unknown</span></span>
<span><span class="co">## 2       F513                                                     Unknown</span></span>
<span><span class="co">## 3      F1016 w/o MS2:3-Hydroxypyridine; CE0; GRFNBEZIAWKNCO-UHFFFAOYSA-N</span></span>
<span><span class="co">## 4      F1279                   w/o MS2:1-AMINOCYCLOPROPANE-1-CARBOXYLATE</span></span>
<span><span class="co">## 5      F1428                         w/o MS2:L-2,3-DIAMINOPROPIONIC ACID</span></span>
<span><span class="co">## 6      F1686                                                     Unknown</span></span>
<span><span class="co">##     MSDialSMILES    rt        mz   ID                               Name</span></span>
<span><span class="co">## 1           null 1.260  76.03783  L30                            Glycine</span></span>
<span><span class="co">## 2           null 1.148  76.07460 L631             Trimethylamine N-oxide</span></span>
<span><span class="co">## 3    c1cc(cnc1)O 1.324  96.04393 L804                  2-Hydroxypyridine</span></span>
<span><span class="co">## 4 NC1(CC1)C(O)=O 0.756 102.05508 L659 1-Aminocyclopropanecarboxylic acid</span></span>
<span><span class="co">## 5   NCC(N)C(O)=O 6.306 105.06686 L936          2,3-Diaminopropionic acid</span></span>
<span><span class="co">## 6           null 1.356 110.05829 L741                      2-Aminophenol</span></span>
<span><span class="co">##   PubChemCID</span></span>
<span><span class="co">## 1        750</span></span>
<span><span class="co">## 2       1145</span></span>
<span><span class="co">## 3       8871</span></span>
<span><span class="co">## 4        535</span></span>
<span><span class="co">## 5      97328</span></span>
<span><span class="co">## 6       5801</span></span></code></pre>
<p>The spectra from all the features that match a metabolite can be
inspected creating the following plot through:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize_associated_spectra.html">visualize_associated_spectra</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm_bio</span>, mR_library <span class="op">=</span> <span class="va">mR_library</span>, metabolite_id <span class="op">=</span> <span class="st">"L1660"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/L1660_F10165_M1028_mirrored.jpg" style="width:80.0%"></p>
<p>The function <code><a href="../reference/h_map_MSMS_comparison.html">h_map_MSMS_comparison()</a></code> draws heatmaps to
visually compare ppm errors and RI differences between feature and
metabolite spectra:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/h_map_MSMS_comparison.html">h_map_MSMS_comparison</a></span><span class="op">(</span><span class="va">mL_norm_bio</span>, metab_id <span class="op">=</span> <span class="st">"L1660"</span>, feature_id <span class="op">=</span> <span class="st">"F10165"</span><span class="op">)</span></span></code></pre></div>
<p><img src="images/ppm_error_L1660_M1028_F10165.jpg" style="width:80.0%"><img src="images/RI_diff_L1660_M1028_F10165.jpg" style="width:80.0%"></p>
</div>
<div class="section level2">
<h2 id="retriving-statistics-for-identified-metabolites">Retriving statistics for identified metabolites<a class="anchor" aria-label="anchor" href="#retriving-statistics-for-identified-metabolites"></a>
</h2>
<p>The results of statistical test and metabolite identification can be
merged using the function <code><a href="../reference/merge_stats_with_features.html">merge_stats_with_features()</a></code>. The
argument <code>feature_stats</code> should be the name of any
statistical test saved in the <code>mRList</code> or a custom data frame
with Feature_ID as row names:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metab_stat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_stats_with_features.html">merge_stats_with_features</a></span><span class="op">(</span>mRList <span class="op">=</span> <span class="va">mL_norm_bio</span>, feature_stats <span class="op">=</span> <span class="st">"anova"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     Feature_ID                                                   MSDialName</span></span>
<span><span class="co">## 191      F3957                                 Lysine; LC-ESI-QTOF; MS2; CE</span></span>
<span><span class="co">## 287      F9507 w/o MS2:"NCGC00160264-01!6,7-Dimethoxy-quinazoline-2,4-diol"</span></span>
<span><span class="co">## 147      F3081      w/o MS2:5-Methyl-1H-benzotriazole; LC-ESI-ITFT; MS2; CE</span></span>
<span><span class="co">## 126      F2833                                               Pipecolic acid</span></span>
<span><span class="co">## 75       F2141                                                      Unknown</span></span>
<span><span class="co">## 216      F4671                                                      Unknown</span></span>
<span><span class="co">##                        MSDialSMILES        ID               Name PubChemCID</span></span>
<span><span class="co">## 191                 NCCCCC(N)C(O)=O       L43           L-Lysine       5962</span></span>
<span><span class="co">## 287 O=C1NC(=O)C2=CC(OC)=C(OC)C=C2N1      L774    L-Cystathionine     439258</span></span>
<span><span class="co">## 147            CC1=CC2=C(C=C1)N=NN2     L1839    5-Hydroxyindole      16054</span></span>
<span><span class="co">## 126                  O=C(O)C1NCCCC1      L377   L-Pipecolic acid     439227</span></span>
<span><span class="co">## 75                             null L171;L504   Betaine;L-Valine   247;6287</span></span>
<span><span class="co">## 216                            null     L1644 N-Acetyl-L-proline      66141</span></span>
<span><span class="co">##             F            p            q      DD-AA     MM-AA     MM-DD</span></span>
<span><span class="co">## 191 206.01628 7.463065e-32 2.164289e-29 0.01168747 0.0000000 0.0000000</span></span>
<span><span class="co">## 287 146.11859 4.178003e-27 6.058104e-25 0.00000000 0.9405020 0.0000000</span></span>
<span><span class="co">## 147 133.75759 6.186876e-26 5.980647e-24 0.07287089 0.0000000 0.0000000</span></span>
<span><span class="co">## 126 126.43314 3.351518e-25 2.429851e-23 0.83474546 0.0000000 0.0000000</span></span>
<span><span class="co">## 75   86.56099 1.571034e-20 6.508571e-19 0.00000000 0.0000000 0.7251425</span></span>
<span><span class="co">## 216  84.48792 3.007217e-20 1.090116e-18 0.00000000 0.0431562 0.0000000</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="pathway-analysis">Pathway analysis<a class="anchor" aria-label="anchor" href="#pathway-analysis"></a>
</h2>
<p>margheRita implements both Over Representation Analysis (ORA) and
Metabolite Set Enrichment Analysis (MSEA), based on clusterProfiler
<span class="citation">(<a href="#ref-Wu2021" role="doc-biblioref">Wu et
al. 2021</a>)</span> over BioCyc, KEGG and Reactome pathway databases.
These analyses can be run by means of function
<code><a href="../reference/pathway_analysis.html">pathway_analysis()</a></code>, which takes as input a vector of
PubChemCID or a ranked list of PubChemCID in case of ORA or MSEA
respectively:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pa_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathway_analysis.html">pathway_analysis</a></span><span class="op">(</span><span class="va">metab_stat</span>, type <span class="op">=</span> <span class="st">"ora"</span>, universe <span class="op">=</span> <span class="va">metab_universe</span><span class="op">)</span></span></code></pre></div>
<p>The result is a list that contains (i) a table with pathway
descriptions and (ii) an object of class “enrichResult”, which can be
used to obtain various visualizations through clusterProfiler
functions.</p>
<pre><code><span><span class="co">##              ID                               Description GeneRatio BgRatio</span></span>
<span><span class="co">## 1270158 1270158 Metabolism of amino acids and derivatives       5/8  30/134</span></span>
<span><span class="co">## 83035     83035                          ABC transporters       4/8  22/134</span></span>
<span><span class="co">## 790012   790012               Biosynthesis of amino acids       3/8  19/134</span></span>
<span><span class="co">## 83030     83030               Aminoacyl-tRNA biosynthesis       2/8  12/134</span></span>
<span><span class="co">## 172847   172847          Protein digestion and absorption       2/8  13/134</span></span>
<span><span class="co">##             pvalue  p.adjust    qvalue                     geneID Count</span></span>
<span><span class="co">## 1270158 0.01408516 0.2242475 0.2098222 439258/439227/247/586/1123     5</span></span>
<span><span class="co">## 83035   0.02491639 0.2242475 0.2098222          5962/247/1123/750     4</span></span>
<span><span class="co">## 790012  0.08569958 0.5141975 0.4811205            5962/439258/750     3</span></span>
<span><span class="co">## 83030   0.15212213 0.5227684 0.4891401                   5962/750     2</span></span>
<span><span class="co">## 172847  0.17425614 0.5227684 0.4891401                   5962/750     2</span></span></code></pre>
<p><img src="images/ora_barplot.png" style="width:80.0%"></p>
<p>See the <a href="https://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html" class="external-link">documentation</a>
of clusterProfiler for further information.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Castellano2021" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domı́nguez, Francesc
Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021.
<span>“POMAShiny: A User-Friendly Web-Based Workflow for Metabolomics
and Proteomics Data Analysis.”</span> <em>PLOS Computational
Biology</em> 17 (7): e1009148.
</div>
<div id="ref-Dieterle2006" class="csl-entry">
Dieterle, Frank, Alfred Ross, Götz Schlotterbeck, and Hans Senn. 2006.
<span>“Probabilistic Quotient Normalization as Robust Method to Account
for Dilution of Complex Biological Mixtures. Application in 1h NMR
Metabonomics.”</span> <em>Analytical Chemistry</em> 78 (13): 4281–90. <a href="https://doi.org/10.1021/ac051632c" class="external-link">https://doi.org/10.1021/ac051632c</a>.
</div>
<div id="ref-Gu2016" class="csl-entry">
Gu, Zuguang, Roland Eils, and Matthias Schlesner. 2016. <span>“Complex
Heatmaps Reveal Patterns and Correlations in Multidimensional Genomic
Data.”</span> <em>Bioinformatics</em>.
</div>
<div id="ref-Klavus2020" class="csl-entry">
Klåvus, Anton, Marietta Kokla, Stefania Noerman, Ville M. Koistinen,
Marjo Tuomainen, Iman Zarei, Topi Meuronen, et al. 2020.
<span>“<span>‘Notame’</span>: Workflow for Non-Targeted LC–MS Metabolic
Profiling.”</span> <em>Metabolites</em> 10 (4). <a href="https://doi.org/10.3390/metabo10040135" class="external-link">https://doi.org/10.3390/metabo10040135</a>.
</div>
<div id="ref-Stacklies2007" class="csl-entry">
Stacklies, Wolfram, Henning Redestig, Matthias Scholz, Dirk Walther, and
Joachim Selbig. 2007. <span>“pcaMethods – a Bioconductor Package
Providing PCA Methods for Incomplete Data.”</span>
<em>Bioinformatics</em> 23: 1164–67.
</div>
<div id="ref-Tsugawa2015" class="csl-entry">
Tsugawa, Hiroshi, Tomas Cajka, Tobias Kind, Yan Ma, Brendan Higgins,
Kazutaka Ikeda, Mitsuhiro Kanazawa, Jean VanderGheynst, Oliver Fiehn,
and Masanori Arita. 2015. <span>“MS-DIAL: Data-Independent MS/MS
Deconvolution for Comprehensive Metabolome Analysis.”</span> <em>Nature
Methods</em> 12 (June): 523–26. <a href="https://doi.org/10.1038/nmeth.3393" class="external-link">https://doi.org/10.1038/nmeth.3393</a>.
</div>
<div id="ref-Wu2021" class="csl-entry">
Wu, Tianzhi, Erqiang Hu, Shuangbin Xu, Meijun Chen, Pingfan Guo, Zehan
Dai, Tingze Feng, et al. 2021. <span>“clusterProfiler 4.0: A Universal
Enrichment Tool for Interpreting Omics Data.”</span> <em>The
Innovation</em> 2 (3): 100141. <a href="https://doi.org/10.1016/j.xinn.2021.100141" class="external-link">https://doi.org/10.1016/j.xinn.2021.100141</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ettore Mosca, Zahrasadat Aalavikakhki, Edoardo Bellini, Valeria Mannella, Gianfranco Frigerio.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
